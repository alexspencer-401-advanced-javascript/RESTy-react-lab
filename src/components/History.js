import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';

const History = ({ history }) => {
  const historyElements = history.map(thing => {
    <li key={`${thing.url}`}>
      <HistoryItem {...thing} />
    </li>;
  });
  return (
    <>
      <h1>History</h1>
      <ul>
        {historyElements}
      </ul>
    </>
  );
};

History.propTypes = {
  history: PropTypes.array
};

export default History;

