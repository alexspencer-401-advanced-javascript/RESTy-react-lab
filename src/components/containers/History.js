import React, { Component } from 'react';

export default class History extends Component {
  // state = {
  
  // }

  render() {
    return (
      <h1>History</h1>
    );
  }
}
