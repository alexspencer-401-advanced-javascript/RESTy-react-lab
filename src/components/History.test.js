import React from 'react';
import { shallow } from 'enzyme';
import History from './History';

describe('History component', () => {
  it('renders History', () => {
    const wrapper = shallow(<History history={[
      {
        url: 'https://hey-arnold-api.herokuapp.com/api/v1/characters',
        method: 'GET'
      }

    ]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
