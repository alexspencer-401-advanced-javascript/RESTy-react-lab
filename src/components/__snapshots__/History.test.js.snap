// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders History 1`] = `
<Fragment>
  <h1>
    History
  </h1>
  <ul>
    <HistoryItem
      key="0"
      method="GET"
      url="https://hey-arnold-api.herokuapp.com/api/v1/characters"
    />
  </ul>
</Fragment>
`;
